### PR TITLE
Add cors filter to handle preflight and allowed methods

### PR DESCRIPTION
**Pull request by bcullerton**
Create a cors filter to set up preflight options responses and populate and allowed methods response header.

### CODE REVIEW

**Review by VaughanJackson123 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/api/filter/CustomCorsFilter.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import jakarta.servlet.http.HttpServletResponse;
+
+public class CustomCorsFilter implements Filter {
+
+    private List<String> externalMethods;
+    private String OPTIONS_METHOD = "OPTIONS";
+    private String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+    private String ACCESS_CONTROL_ALLOW_HEADERS = "Access-Control-Allow-Headers";
+    private String ACCESS_CONTROL_MAX_AGE = "Access-Control-Max-Age";
+    private String ACCESS_CONTROL_ALLOW_METHODS = "Access-Control-Allow-Methods";
```
Comment (NeilShadrach) [inline src/main/java/uk/gov/companieshouse/api/filter/CustomCorsFilter.java:23]: Could off-the-shelf constants be used for the ACCESS_CONTROL strings?
eg org.springframework.http.HttpHeaders

**Review by NeilShadrach — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Pull request merged**